ValveKeyValue/ValveKeyValue.Test/Text/ConditionalTestCase.cs

```diff
@@ -105,6 +106,52 @@ public void ConditionalInKey()
             Assert.That((string)children[0].Value, Is.EqualTo("windows 32-bit"));
         }
 
+        [Test]
+        public void ConditionalBeforeObject()
```

Comment:
Validated against Valve's tier1 implementation:

```
> steamdb_vkv_test.exe -file conditional_before_object_value.vdf
Name: test case
    operating system (string): windows 32-bit
```

and if I change the condition to `[$OSX]`:

```
> steamdb_vkv_test.exe -file conditional_before_object_value.vdf
Name: test case
```

(i.e. empty object)